crates/red_knot/src/symbols.rs

```diff
@@ -431,12 +439,14 @@ impl SymbolTable {
             }
             RawEntryMut::Vacant(entry) => {
                 let id = self.symbols_by_id.push(Symbol {
-                    name,
+                    name_id,
                     flags,
                     scope_id,
                 });
                 entry.insert_with_hasher(hash, id, (), |symid| {
-                    SymbolTable::hash_name(&self.symbols_by_id[*symid].name)
+                    // FIXME: add `Name.hash(self, Names)` to encapsulate this? or even
+                    // `NameId.hash(self, Names)`?
```

Comment:
If we only do this once, does it need encapsulating?

Reply:
I think it's an internal detail of the hash table that shouldn't be in this module. It's coupled to the implementation of `Names`.